docs/security/apparmor.md

```diff
@@ -97,6 +97,133 @@ $ apparmor_parser -R /path/to/profile
 $ /etc/init.d/apparmor start
 ```
 
+### Resources for writing profiles
+
+The syntax for file globbing in AppArmor is a bit different than some other
+globbing implementations. It is highly suggested you take a look at some of the
+below resources with regard to AppArmor profile syntax.
+
+- [Quick Profile Language](http://wiki.apparmor.net/index.php/QuickProfileLanguage)
+- [Globbing Syntax](http://wiki.apparmor.net/index.php/AppArmor_Core_Policy_Reference#AppArmor_globbing_syntax)
+
+## Examples
+
+### Nginx Custom Profile
+
+Let's create a custom AppArmor profile for nginx to show how this works.
+
+1. Save the custom profile to disk.
+
+Below is our custom profile we will save in `/etc/apparmor.d/containers/docker-nginx`.
+
+> **NOTE:** The path we save to is not a requirement to be the same as above,
+> this was only used as the path for this example.
+
+```
+#include <tunables/global>
+
+
+profile docker-nginx-sample flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+
+
+  network inet tcp,
+  network inet udp,
+  network inet icmp,
+
+  deny network raw,
+
+  deny network packet,
+
+  file,
+  umount,
+
+  deny /bin/** wl,
+  deny /boot/** wl,
+  deny /dev/** wl,
+  deny /etc/** wl,
+  deny /home/** wl,
+  deny /lib/** wl,
+  deny /lib64/** wl,
+  deny /media/** wl,
+  deny /mnt/** wl,
+  deny /opt/** wl,
+  deny /proc/** wl,
+  deny /root/** wl,
+  deny /sbin/** wl,
+  deny /srv/** wl,
+  deny /tmp/** wl,
+  deny /sys/** wl,
+  deny /usr/** wl,
+
+  audit /** w,
+
+  /var/run/nginx.pid w,
+
+  /usr/sbin/nginx ix,
+
+  deny /bin/dash mrwklx,
+  deny /bin/sh mrwklx,
+  deny /usr/bin/top mrwklx,
+
+
+  capability chown,
+  capability dac_override,
+  capability setuid,
+  capability setgid,
+  capability net_bind_service,
+
+  deny @{PROC}/{*,**^[0-9*],sys/kernel/shm*} wkx,
+  deny @{PROC}/sysrq-trigger rwklx,
+  deny @{PROC}/mem rwklx,
+  deny @{PROC}/kmem rwklx,
+  deny @{PROC}/kcore rwklx,
+  deny mount,
+  deny /sys/[^f]*/** wklx,
+  deny /sys/f[^s]*/** wklx,
+  deny /sys/fs/[^c]*/** wklx,
+  deny /sys/fs/c[^g]*/** wklx,
+  deny /sys/fs/cg[^r]*/** wklx,
+  deny /sys/firmware/efi/efivars/** rwklx,
+  deny /sys/kernel/security/** rwklx,
+}
+```
+2. Load the profile.
+
+```
+$ sudo apparmor_parser -r -W /etc/apparmor.d/containers/docker-nginx
+```
+
+3. Run a container with our profile.
+
+```
```

Comment:
ah i fixed it by making them bold :)

On Thu, Apr 7, 2016 at 6:15 PM, Kai Qiang Wu <notifications@github.com>
wrote:

> In docs/security/apparmor.md
> <https://github.com/docker/docker/pull/21861#discussion_r58969301>:
>
> > +  deny /sys/fs/[^c]*/** wklx,
> > +  deny /sys/fs/c[^g]*/** wklx,
> > +  deny /sys/fs/cg[^r]*/** wklx,
> > +  deny /sys/firmware/efi/efivars/** rwklx,
> > +  deny /sys/kernel/security/** rwklx,
> > +}
> > +```
> > +2. Load the profile.
> > +
> > +```
> > +$ sudo apparmor_parser -r -W /etc/apparmor.d/containers/docker-nginx
> > +```
> > +
> > +3. Run a container with our profile.
> > +
> > +```
>
> same as above
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/21861/files/a883e80102ad52747e959b0750e8f79c401176be#r58969301>
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>